plugin.cpp

```diff
@@ -56,13 +56,16 @@ typedef enum
 	INCLUDE,
 	EXCLUDE,
 	RENAME,
+	REMOVE,
 	DPMAP
 } action;
 
 struct AssetAction {
 	action actn;
 	string new_asset_name; // valid only in case of RENAME
 	map<string, string> dpmap;
+	string datapoint; // valid only in case of REMOVE
+	string type; // valid only in case of REMOVE
```

Comment:
You didn't use mutex during reconfigure / ingest. is it safe to ingest and reconfigure simultaneously.

Reply:
fixed